packages/fx-core/src/core/FxCoreImplementV3.ts

```diff
@@ -364,7 +366,17 @@ export class FxCoreV3Implement {
         return err(new InvalidProjectError());
       }
       const trackingId = await getTrackingIdFromPath(projectPath);
-      const isSupport = getVersionState(versionInfo);
+      let isSupport: VersionState;
+      // As projectVersionCheck is a v3 interface, v3 not enabled case is an exception and only called by vs platform
+      if (!isV3Enabled() && inputs.platform === Platform.VS) {
```

Comment:
I think it is also correct when the platform is not VS.

Reply:
Yes. Currently only VS platform will use the interface in v2.
If other platforms use the interface in v2, we should check the flow.